test_qmk_compiler.py

```diff
@@ -0,0 +1,337 @@
+
+
+def test_0014_repo_name_qmk_firmware():
+    """Make sure that the qmk_firmware git url is reliably turned into qmk_firmware.
+    """
+    assert qmk_commands.repo_name('https://github.com/qmk/qmk_firmware.git') == 'qmk_firmware'
+
+
+def test_0015_repo_name_chibios():
+    """Make sure that the qmk_firmware git url is reliably turned into qmk_firmware.
+    """
+    assert qmk_commands.repo_name('https://github.com/qmk/ChibiOS.git') == 'chibios'
+
+
+def test_0016_repo_name_chibios_contrib():
+    """Make sure that the qmk_firmware git url is reliably turned into qmk_firmware.
+    """
+    assert qmk_commands.repo_name('https://github.com/qmk/ChibiOS-Contrib.git') == 'chibios-contrib'
+
+
+def test_0017_find_all_layouts_cluecard():
+    """Make sure that update_kb_redis.find_all_layouts() can find the cluecard layout.
+    """
+    layouts = update_kb_redis.find_all_layouts('clueboard/card')
+    assert list(layouts) == ['KEYMAP']
+    assert layouts['KEYMAP'] == {
+        'key_count': 12,
+        'layout': [
+            {'label': 'k00', 'w': 1, 'x': 0, 'y': 0},
+            {'label': 'k01', 'w': 1, 'x': 1, 'y': 0},
+            {'label': 'k02', 'w': 1, 'x': 2, 'y': 0},
+            {'label': 'k10', 'w': 1, 'x': 0, 'y': 1},
+            {'label': 'k12', 'w': 1, 'x': 1, 'y': 1},
+            {'label': 'k20', 'w': 1, 'x': 0, 'y': 2},
+            {'label': 'k21', 'w': 1, 'x': 1, 'y': 2},
+            {'label': 'k22', 'w': 1, 'x': 2, 'y': 2},
+            {'label': 'k11', 'w': 1, 'x': 0, 'y': 3},
+            {'label': 'k30', 'w': 1, 'x': 0, 'y': 4},
+            {'label': 'k31', 'w': 1, 'x': 1, 'y': 4},
+            {'label': 'k32', 'w': 1, 'x': 2, 'y': 4}
+        ]
+    }
+
+
+### FIXME: Need to write a test for update_kb_redis.parse_config_h()
+
+
+def test_0019_parse_config_h_file_cluecard():
+    """Make sure the config.h parsing works.
+    """
+    config_h = update_kb_redis.parse_config_h_file('qmk_firmware/keyboards/clueboard/card/config.h')
+    assert config_h == {
+        'CONFIG_H': True,
+        'VENDOR_ID': '0xC1ED',
+        'PRODUCT_ID': '0x2330',
+        'DEVICE_VER': '0x0001',
+        'MANUFACTURER': 'Clueboard',
+        'PRODUCT': 'ATMEGA32U4 Firmware Dev Kit',
+        'DESCRIPTION': 'A small board to help you hack on QMK.',
+        'MATRIX_ROWS': '4',
+        'MATRIX_COLS': '3',
+        'MATRIX_ROW_PINS': '{ F0, F5, F4, B4 }',
+        'MATRIX_COL_PINS': '{ F1, F7, F6 }',
+        'UNUSED_PINS': True,
+        'DIODE_DIRECTION': 'ROW2COL',
+        'DEBOUNCING_DELAY': '20',
+        'BACKLIGHT_LEVELS': '6',
+        'IS_COMMAND()': '( \\',
+        'RGB_DI_PIN': 'E6',
+        'RGBLED_NUM': '4 // Number of LEDs',
+        'RGBLIGHT_HUE_STEP': '10',
+        'RGBLIGHT_SAT_STEP': '17',
+        'RGBLIGHT_VAL_STEP': '17'
+    }
+
+
+## FIXME: Write a test for update_kb_redis.parse_rules_mk
+
+
+def test_0021_parse_rules_mk_file_cluecard():
+    """Make sure the rules.mk parsing works.
```

Comment:
I think these tests are very valuable, might want to use a table driven method though so we can add a couple more keyboards easily.

Reply:
Longer term I want to isolate these test cases from qmk_firmware. Have a few different test cases that test comprehensively rather than rely on keyboards inside qmk_firmware.